flow/IThreadPoolTest.actor.cpp

```diff
@@ -0,0 +1,61 @@
+#include "flow/IThreadPool.h"
+
+#include <pthread.h>
+#include <ostream>
+
+#include "flow/UnitTest.h"
+#include "flow/actorcompiler.h" // has to be last include
+
+// Thread naming only works on Linux.
+#if defined(__linux__)
+
+void forceLinkIThreadPoolTests() {}
+
+struct ThreadNameReceiver : IThreadPoolReceiver {
+	void init() override {}
+
+	struct GetNameAction : TypedAction<ThreadNameReceiver, GetNameAction> {
+		ThreadReturnPromise<std::string> name;
+
+		double getTimeEstimate() const override { return 3.; }
+	};
+
+	void action(GetNameAction& a) {
+		pthread_t t = pthread_self();
+		const size_t arrayLen = 16;
+		char name[arrayLen];
+		int err = pthread_getname_np(t, name, arrayLen);
+		if (err != 0) {
+			std::cout << "Get name filed with error code: " << err << std::endl;
+			a.name.sendError(unknown_error());
```

Comment:
Unknown error is typically used for cases where a non-flow exception was caught by an actor, etc. Platform error is perhaps more appropriate here:

```suggestion
			a.name.sendError(platform_error());
```

you can also add `.GetLastError()` if this function sets `errno`.

Reply:
My reading of the pthread documentation is that it does not set `errno`, but that's great to know for the future!